src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs

```diff
@@ -272,9 +272,14 @@ private void PerformPackageUpdatesOrReinstalls()
             foreach (var project in Projects)
             {
                 var installedPackages = await project.GetInstalledPackagesAsync(Token);
-                if (installedPackages.Select(installedPackage => installedPackage.PackageIdentity.Id)
-                                     .Any(installedPackageId => installedPackageId.Equals(packageId, StringComparison.OrdinalIgnoreCase)))
+
+                var installedPackage = installedPackages.Where(
```

Comment:
Its actually single enumeration with .Any() since .First() will not enumerate the whole list. But yes, FirstOrDefault() makes more sense. Will udpate it.